lib/ipam/ipam.go

```diff
@@ -744,6 +735,16 @@ func (c ipamClient) releaseIPsFromBlock(ctx context.Context, ips []net.IP, block
 		for handleID, amount := range handles {
 			c.decrementHandle(ctx, handleID, blockCIDR, amount)
 		}
+
+		// Determine whether or not the block's pool still matches the node.
+		if _, err := c.ensureConsistentAffinity(ctx, obj.Value.(*model.AllocationBlock)); err != nil {
+			if updateErr, ok := err.(cerrors.ErrorResourceUpdateConflict); ok {
```

Comment:
Hm, I think this is probably not going to work the way we expect any more now that it is moved.

Let's say we hit a conflict error - when we retry above, the block will have already released the IPs and so we'll probably hit line 701 and return without reaching this code a second time.

Similarly, I think if we hit some other error, we'll return an error to the caller indicating we couldn't release the IPs when in fact we _did_ actually release the IPs.

We probably need to rethink what the right thing to do on failure is. WDYT?